server/repository/src/db_diesel/name_insurance_join_row.rs

```diff
@@ -1,12 +1,29 @@
 use super::{
-    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, RowActionType, StorageConnection,
+    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, EqualFilter, RowActionType,
+    StorageConnection,
 };
 
-use crate::{repository_error::RepositoryError, Upsert};
+use crate::{
+    diesel_macros::{apply_equal_filter, apply_sort},
+    repository_error::RepositoryError,
+    Sort, Upsert,
+};
 use diesel::prelude::*;
 use diesel_derive_enum::DbEnum;
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, Default, PartialEq, Debug)]
+pub struct NameInsuranceJoinFilter {
+    pub insurance_provider_id: Option<EqualFilter<String>>,
+}
+
+pub enum NameInsuranceJoinSortField {
+    ExpiryDate,
+    IsActive,
+}
+
+pub type NameInsuranceJoinSort = Sort<NameInsuranceJoinSortField>;
+
 #[derive(DbEnum, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
 #[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 #[DbValueStyle = "SCREAMING_SNAKE_CASE"]
```

Comment:
I think this is probably a good approach for joining to name but not needing to return it...

https://github.com/msupply-foundation/open-msupply/compare/6417-patient-insurance-graphql-and-service-layer...6417-name-link-suggestion